simpleclient_spring_web/src/main/java/io/prometheus/client/spring/resttemplate/MetricsClientHttpRequestInterceptor.java

```diff
@@ -0,0 +1,63 @@
+package io.prometheus.client.spring.resttemplate;
+
+import io.prometheus.client.Summary;
+import io.prometheus.client.Summary.Child;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+public class MetricsClientHttpRequestInterceptor implements ClientHttpRequestInterceptor {
+
+  private static final List<MetricsRestTemplateLabelType> DEFAULT_LABELS =
+      Arrays.asList(MetricsRestTemplateLabelType.values());
+
+  private final Summary metricClientInterceptorSummary;
+
+  private final List<MetricsRestTemplateLabelType> labelTypes;
+
+  public MetricsClientHttpRequestInterceptor() {
+    this(DEFAULT_LABELS);
+  }
+
+  public MetricsClientHttpRequestInterceptor(List<MetricsRestTemplateLabelType> labelTypes) {
+    this.labelTypes = labelTypes;
+    metricClientInterceptorSummary = Summary.build()
+        .name("rest_template_request_duration_seconds")
```

Comment:
It's not quite obvious from this name what this means, are these templates being served back to the user?

Something more like spring_rest_client_template_request_duration_seconds would be clearer.